openpype/hosts/blender/plugins/publish/collect_current_file.py

```diff
@@ -34,39 +34,28 @@ def process(self, context):
             "Current file is empty. Save the file before continuing."
         )
 
-        folder, file = os.path.split(current_file)
-        filename, ext = os.path.splitext(file)
-
-        task = legacy_io.Session["AVALON_TASK"]
-
-        data = {}
+        _, file = os.path.split(current_file)
+        filename, _ = os.path.splitext(file)
 
         # create instance
         instance = context.create_instance(name=filename)
+        task = legacy_io.Session["AVALON_TASK"]
```

Comment:
Please replace all `legacy_io` usages. During publishing can be used `context.data["projectName"]`, `context.data["asset"]` and `context.data["task"]`.

In other cases (e.g. in creators) can be used functions from openpype pipeline `get_current_project_name`, `get_current_asset_name` and `get_current_task_name`.